rita_common/src/blockchain_oracle/mod.rs

```diff
@@ -195,7 +195,15 @@ async fn update_blockchain_info(our_address: Address, web3: Web3, full_node: Str
         join4(balance, nonce, net_version, gas_price).await;
     let mut settings = settings::get_rita_common();
     match balance {
-        Ok(balance) => update_balance(&full_node, &mut settings.payment.balance, balance),
+        Ok(balance) => update_balance(
+            &full_node,
+            &mut settings
```

Comment:
this won't work the way you expect it to. modify update balance to return a value and set settings.payment.balance in this scope.

The issue is your passing a mutable reference, but if its None you'll just pass a local var, it will be modifed and the result will not propogate into the settings struct.